src/test/scala-2.11/markets/participants/MarketParticipantLikeSpec.scala

```diff
@@ -68,16 +71,52 @@ class MarketParticipantLikeSpec extends TestKit(ActorSystem("MarketParticipantLi
 
     scenario("A MarketParticipantLike actor receives a Canceled message...") {
 
+      val marketParticipant = TestActorRef(new TestMarketParticipant)
+
+      Given("A MarketParticipantLike actor with outstanding orders...")
+      val order = LimitAskOrder(marketParticipant, 10, 100, 1, tradable, uuid)
+      val accepted = Accepted(order, 2, uuid)
+      marketParticipant ! accepted
+
       When("A MarketParticipantLike actor receives a Canceled message...")
       val canceled = Canceled(order, 3, uuid)
       marketParticipant ! canceled
 
-      Then("...it should remove the canceled orders UUID from its outstanding orders.")
+      Then("...it should remove the canceled order from its outstanding orders.")
       val marketParticipantActor = marketParticipant.underlyingActor
       marketParticipantActor.outstandingOrders.headOption should be(None)
 
     }
 
+    scenario("A MarketParticipantLike actor receives a Filled message...") {
+
+      val marketParticipant = TestActorRef(new TestMarketParticipant)
+
+      Given("A MarketParticipantLike actor with outstanding orders...")
+      val order1 = LimitAskOrder(marketParticipant, 10, 100, 1, tradable, uuid)
+      val order2 = MarketBidOrder(marketParticipant, 1000, 2, tradable, uuid)
+      val acceptedOrders = immutable.Seq(Accepted(order1, 3, uuid), Accepted(order2, 3, uuid))
+      acceptedOrders.foreach(order => marketParticipant ! order)
+
+      When("A MarketParticipantLike actor receives a Filled message with no residual order...")
+      val filled = Filled(order1, None, 4, uuid)
+      marketParticipant ! filled
+
+      Then("...it should remove the filled order from its outstanding orders.")
+      val marketParticipantActor = marketParticipant.underlyingActor
+      marketParticipantActor.outstandingOrders.headOption should be(Some(order2))
+
+      When("A MarketParticipantLike actor receives a Filled message with some residual order...")
+      val(_, residualOrder) = order2.split(500)
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Field name does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/davidrpugh/markets-sandbox/file/1549988767/issues/source?bid=2921361&fileBranchId=3014639#l110)